src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldLikeEventSymbol.cs

```diff
@@ -144,14 +153,14 @@ public override TypeWithAnnotations TypeWithAnnotations
             get { return _type; }
         }
 
-        public override MethodSymbol AddMethod
+        public override MethodSymbol? AddMethod
         {
-            get { return _addMethod; }
+            get { return PartialImplementationPart?.AddMethod ?? _addMethod; }
```

Comment:
Something feels off about this to me. Partial property accessors are themselves represented as partial methods in the symbol model. It feels most straightforward for partial event accessors to work in a similar way, even though the accessors on the definition part are not "addressable", e.g. user cannot put attributes on them, and so on.

Mainly I am concerned about consistency of the symbol model, e.g., someone might expect to say `eventSymbol.AddMethod.ContainingSymbol` and get `eventSymbol` as the answer.

Reply:
> Mainly I am concerned about consistency of the symbol model, e.g., someone might expect to say `eventSymbol.AddMethod.ContainingSymbol` and get `eventSymbol` as the answer.

That makes sense. But I don't know how to fix this - do you think we should create a "duplicate" symbol for the same event accessor?

Reply:
Yeah I think you would make a distinct accessor symbol on the definition part, which is a partial method, and let it plunk through the existing partial method machinery, much like partial property accessors do.

But, before moving forward with that, it might be good to raise it with the broader compiler team. @AlekseyTs in case you have any thoughts here.